packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -42,11 +46,32 @@ function useIsPostTypeHierarchical( postType ) {
 	);
 }
 
+function useAllowedControls( attributes ) {
+	return useSelect(
+		( select ) =>
+			select( blocksStore ).getActiveBlockVariation(
+				queryLoopName,
+				attributes
+			)?.allowControls || EMPTY_ARRAY,
+
+		[ attributes ]
+	);
+}
+
+function isControllAllowed( allowedControls, key ) {
+	// Every controls is allowed if the list is empty or not defined.
+	if ( ! allowedControls?.length ) {
```

Comment:
Wouldn't it be best to just check for undefined? What if a user wants to disable all native controls?

Reply:
It does check for undefined(`?.`) and also handles an empty array this way..

Reply:
Yea, what I am trying to say is that perhaps it should _just_ check for undefined, because empty array could be a valid user choice (i.e. disable everything)

Reply:
Oh, got it now. Do you think there are use cases we need that? I can update..

Reply:
That would be my suggestion. There can be a situation in which all core controls would need to be disabled, and I suggest we don't disallow that _a priori_.